components/Note/index.js

```diff
@@ -0,0 +1,105 @@
+import styled from "styled-components";
+import useSWR from "swr";
+import { useState } from "react";
+import NotesForm from "../NotesForm";
+import { useRouter } from "next/router";
+
+const StyledListItem = styled.li`
+  background-color: rgba(255, 255, 255, 0.6);
+  padding: 10px;
+  display: flex;
+  flex-direction: column;
+  align-content: center;
+`;
+
+export default function Note({ note, locatData }) {
+  const router = useRouter();
+  const { mutate } = useSWR(`/api/locations/${locatData?._id}`);
+  const [isEditMode, setIsEditMode] = useState(false);
+
+  async function handleEditNote(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const noteData = Object.fromEntries(formData);
+
+    const response = await fetch(`/api/notes/${note?._id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(noteData),
+    });
+    if (response.ok) {
+      mutate();
+      setIsEditMode(false);
+    }
+  }
+  async function handleDeleteNote() {
```

Comment:
The same is true of the delete function. Move it up and keep components as dumb as possible.